server/src/game/role/marksman.rs

```diff
@@ -0,0 +1,208 @@
+
+use std::ops::Deref;
+
+use serde::Serialize;
+
+use crate::game::chat::{ChatGroup, ChatMessageVariant};
+use crate::game::grave::GraveKiller;
+use crate::game::phase::PhaseType;
+use crate::game::player::PlayerReference;
+use crate::game::role_list::Faction;
+use crate::game::visit::Visit;
+
+use crate::game::Game;
+use super::{Priority, RoleStateImpl, Role, RoleState};
+
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Marksman {
+    state: MarksmanState
+}
+
+#[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "type")]
+pub(self) enum MarksmanState{
+    NotLoaded,
+    Marks{
+        marks: MarksmanMarks
+    },
+    ShotTownie
+}
+impl MarksmanState{
+    fn no_marks(&self)->bool{
+        !matches!(
+            self,
+            MarksmanState::Marks{marks: MarksmanMarks::One{..}} |
+            MarksmanState::Marks{marks: MarksmanMarks::Two{..}}
+        )
```

Comment:
How about matches None?

Reply:
There are more cases i would have to handle there
No marks, not loaded, and shot townie. This works